src/convertToObject.js

```diff
@@ -7,6 +7,22 @@
  */
 function convertToObject(sourceString) {
   // write your code here
+  const prop = sourceString
+    .split(';')
+    .map((el) => el.trim())
+    .filter((el) => el !== '');
+
+  const object = {};
+
+  prop.forEach((item) => {
+    const [key, value] = item.split(':').map((part) => part.trim());
```

Comment:
The code assumes that each item will contain a colon to split into key-value pairs. Consider adding error handling or validation to manage cases where the colon might be missing, which would result in unexpected behavior.